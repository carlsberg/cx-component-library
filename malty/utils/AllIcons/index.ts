export { allIcons, allIconsStoryOptions } from './AllIconsMap';
