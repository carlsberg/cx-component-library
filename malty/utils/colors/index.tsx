const rgbToHex = (value: string): string =>
  value.replace(/rgb\((\d+),\s*(\d+),\s*(\d+)\)/gi, (_, r, g, b) => {
    const toHex = (n: string) => parseInt(n, 10).toString(16).padStart(2, '0');
    return `#${toHex(r)}${toHex(g)}${toHex(b)}`;
  });

export { rgbToHex };
