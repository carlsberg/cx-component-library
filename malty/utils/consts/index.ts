export enum EventKeys {
  ENTER = 'Enter',
  SPACE = 'Space',
  ESCAPE = 'Escape'
}
