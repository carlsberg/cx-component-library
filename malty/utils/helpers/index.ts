export const getBreakpointNumber = (breakpoint: string) => Number(breakpoint.split('px')[0]);
