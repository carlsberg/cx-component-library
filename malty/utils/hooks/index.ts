export { Device, useMatchMedia } from './useMatchMedia';
