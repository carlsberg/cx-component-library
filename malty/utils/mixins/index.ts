import { css } from 'styled-components';

// Hide an element to all devices except screen readers. More info here: https://webaim.org/techniques/css/invisiblecontent/
export const srOnly = css`
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
`;
