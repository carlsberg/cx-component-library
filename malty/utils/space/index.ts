import type { ArgTypes } from '@storybook/react';

export interface SpaceProps {
  m?: string;
  mt?: string;
  mr?: string;
  mb?: string;
  ml?: string;
  mx?: string;
  my?: string;
}

export const space = <Props extends SpaceProps>({ m = '', mx, my, mt, mr, mb, ml }: Props) => `
    margin: ${m};
    margin-top: ${mt ?? my ?? ''};
    margin-bottom: ${mb ?? my ?? ''};
    margin-right: ${mr ?? mx ?? ''};
    margin-left: ${ml ?? mx ?? ''};
`;

export function spreadSpaceProps<Props extends SpaceProps>(props: Props) {
  const { m, mt, mr, mb, ml, mx, my, ...restProps } = props;

  return { spaceProps: { m, mt, mr, mb, ml, mx, my }, restProps };
}

export function generateStorybookSpacing(): ArgTypes {
  return {
    m: {
      description: 'margin',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mt: {
      description: 'margin-top',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mr: {
      description: 'margin-right',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mb: {
      description: 'margin-bottom',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    ml: {
      description: 'margin-left',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mx: {
      description: 'margin-left, margin-right',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    my: {
      description: 'margin-top, margin-bottom',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    }
  };
}
