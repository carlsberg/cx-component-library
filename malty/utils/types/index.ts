export interface SpaceProps {
  m?: string;
  mt?: string;
  mr?: string;
  mb?: string;
  ml?: string;
  mx?: string;
  my?: string;
}
