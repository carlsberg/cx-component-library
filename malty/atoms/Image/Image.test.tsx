import { render } from '@carlsberggroup/malty.utils.test';
import { screen } from '@testing-library/react';
import React from 'react';
import { Image } from '.';

const testImageSrc = 'https://via.placeholder.com/400';
const altText = 'Image Component';
const dataTestId = 'image';

describe('Image component', () => {
  it('should renders the component with the correct alt description', () => {
    render(<Image src={testImageSrc} cover={false} alt={altText} dataTestId={dataTestId} />);

    expect(screen.getAllByAltText(altText)).toBeDefined();
  });

  it('should render with the assigned dataTestid', () => {
    render(<Image src={testImageSrc} cover={false} alt={altText} dataTestId={dataTestId} />);

    expect(screen.getByTestId(dataTestId)).toBeInTheDocument();
  });

  it('should render with the correct caption', () => {
    render(<Image src={testImageSrc} cover={false} alt={altText} dataTestId={dataTestId} figcaption="caption text" />);

    expect(screen.getByTestId('image-figcaption')).toHaveTextContent('caption text');
  });

  it('should render with the assigned height', () => {
    render(<Image src={testImageSrc} cover={false} alt={altText} dataTestId={dataTestId} height="50" />);

    expect(screen.getByTestId(dataTestId)).toHaveAttribute('height', '50');
  });

  it('should render with the assigned width', () => {
    render(<Image src={testImageSrc} cover={false} alt={altText} dataTestId={dataTestId} width="50" />);

    expect(screen.getByTestId(dataTestId)).toHaveAttribute('width', '50');
  });
});
