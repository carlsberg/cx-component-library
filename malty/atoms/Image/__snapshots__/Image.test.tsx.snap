// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component matches snapshot 1`] = `
<div
  className="sc-dkzDqf cmapVJ"
>
  <div
    className="sc-hKMtZM dYgOXo"
  />
  <img
    className="sc-gsnTZi DdlAW"
    src="https://via.placeholder.com/400"
  />
</div>
`;
